ui/components/src/ValueMappingEditor/ValueMappingsEditor.tsx

```diff
@@ -0,0 +1,81 @@
+// Copyright 2024 The Perses Authors
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import { Button, Divider, Stack, Typography } from '@mui/material';
+import Grid from '@mui/material/Unstable_Grid2';
+
+import { useState } from 'react';
+import AddIcon from 'mdi-material-ui/Plus';
+import { ValueMapping } from '@perses-dev/core';
+import { ValueMappingEditor } from './ValueMappingEditor';
+
+export interface ValueMappingsEditorProps {
+  mappings: ValueMapping[];
+  onChange: (valueMappings: ValueMapping[]) => void;
+}
```

Comment:
Would be cool if table cell settings could use this mapping editor, in order to not have code duplication. AFAIK only colors change (2 instead of 1).
I propose something like this:
```suggestion
  hasSecondaryColor?: bool
}
```

If it's enabled, we could use "Primary color" and "Secondary color", so it can be used everywhere 🤔 Or if it's to abstract to users and we could add an option to rename headers:

```suggestion
  hasSecondaryColor?: bool
  colorHeaders?: string[]
}
```

Reply:
I agree, and it was my original plan, but since there are options on how to implement it, I would discuss it with others. 
I would move tables to value mapping as well since it is a successor of cell settings and overall functionality duplication. To keep the value mapping API consistent across all panels, I would take inspiration from Grafana, where they let you select one color but also let you choose where it is going to be applied. If it is text color, it applies as is, and if it is background, it calculates the text color for perfect contrast.